### PR TITLE
Merge `Gist` type stub back to source

### PR DESCRIPTION
**Pull request by trim21**

### CODE REVIEW

File: github/Gist.py [hunk 1]
```diff
@@ -214,106 +185,75 @@ def create_comment(self, body):
...
         :calls: `PATCH /gists/{id} <https://docs.github.com/en/rest/reference/gists>`_
-        :param description: string
-        :param files: dict of string to :class:`github.InputFileContent.InputFileContent`
-        :rtype: None
         """
-        assert description is github.GithubObject.NotSet or isinstance(description, str), description
-        assert files is github.GithubObject.NotSet or all(
+        assert is_optional(description, str), description
+        # limitation of `TypeGuard`
+        assert (not is_defined(files)) or all(
```
Comment (trim21) [inline github/Gist.py:207]: `is_undefined` here can't narrow `files` to `dict`, so  we have to use `not isdefined(files)` or `isinstance(files, _NotSetType)`

File: github/Gist.py [hunk 2]
```diff
@@ -30,182 +30,153 @@
...
+    def files(self) -> dict[str, github.GistFile.GistFile]:
         self._completeIfNeeded()
         return self._files.value
 
     @property
-    def fork_of(self):
-        """
-        :type: :class:`github.Gist.Gist`
-        """
+    def fork_of(self) -> github.Gist.Gist:
```
Comment (EnricoMi) [inline github/Gist.py:113]:
```suggestion
    def fork_of(self) -> Gist:
```

File: github/Gist.py [hunk 3]
```diff
@@ -30,182 +30,153 @@
...
+    def id(self) -> str:
         self._completeIfNotSet(self._id)
         return self._id.value
 
     @property
-    def owner(self):
-        """
-        :type: :class:`github.NamedUser.NamedUser`
-        """
+    def owner(self) -> github.NamedUser.NamedUser:
```
Comment (EnricoMi) [inline github/Gist.py:153]:
```suggestion
    def owner(self) -> NamedUser:
```

File: github/Gist.py [hunk 4]
```diff
@@ -30,182 +30,153 @@
...
+    def url(self) -> str:
         self._completeIfNotSet(self._url)
         return self._url.value
 
     @property
-    def user(self):
-        """
-        :type: :class:`github.NamedUser.NamedUser`
-        """
+    def user(self) -> github.NamedUser.NamedUser:
```
Comment (EnricoMi) [inline github/Gist.py:173]:
```suggestion
    def user(self) -> NamedUser:
```

**Review by EnricoMi — CHANGES_REQUESTED** [review]
minor cleanups

**Comment by EnricoMi** [comment]
@trim21 what do you think about my comments above? I'd like to continue merging these PRs and this affects many of your PRs.

**Comment by trim21** [comment]
> @trim21 what do you think about my comments above? I'd like to continue merging these PRs and this affects many of your PRs.

see https://github.com/PyGithub/PyGithub/pull/2727#issuecomment-1684146096

**Review by EnricoMi — APPROVED** [review]
LGTM!

**Pull request merged**